src/sentry/options/store.py

```diff
@@ -145,20 +145,24 @@ def get_store(self, key, silent=False):
 
         Returns None in both cases, if the key doesn't actually exist,
         or if we errored fetching it.
-
-        NOTE: This behavior should probably be improved to differentiate
-        between a miss vs error, but not worth it now since the value
-        is limited at the moment.
         """
+        value = None
+        err = None
         try:
             value = self.model.objects.get(key=key.name).value
-        except (self.model.DoesNotExist, ProgrammingError, OperationalError):
-            value = None
-        except Exception:
+        except (ProgrammingError, OperationalError) as exc:
+            err = exc
```

Comment:
Should we be logging the exception in this case?

Reply:
I guess now that DoesNotExists is a separate branch I can just let it be caught by the base Exception.